src/models/athb.js

```diff
@@ -0,0 +1,256 @@
+ *
+ * Note: vr is the relative air speed caused by body movement and not the air
+ * speed measured by the air speed sensor. The relative air speed is the sum of the
+ * average air speed measured by the sensor plus the activity-generated air speed
+ * (Vag). Where Vag is the activity-generated air speed caused by motion of
+ * individual body parts. vr can be calculated using the function
+ * jsthermalcomfort.utilities.v_relative.
+ *
+ * @param { number[] } rh - relative humidity, [%]
+ * @param { number[] } met - metabolic rate, [met]
+ * @param { number[] } t_running_mean - running mean temperature, in [°C]
+ *
+ * The running mean temperature can be calculated using the function
+ * jsthermalcomfort.utilities.running_mean_outdoor_temperature.
+ *
+ * @returns { number[] } athb_pmv - Predicted Mean Vote calculated with the Adaptive Thermal Heat Balance framework
+ *
+ * @example
+ * // JavaScript code example
+ *
+ * // Import the athb_array function from the jsthermalcomfort module
+ * import { athb_array } from "../../src/models/athb.js";
+ *
+ * // Calculate PMV with sample data when it is arrays.
+ * console.log(athb({ tdb: [25, 27], tr: [25, 25], vr: [0.1, 0.1], rh: [50, 50], met: [1.1, 1.1], t_running_mean: [20, 20] }));
+ * // Output: [0.2, 0.209]
+ */
+export function athb_array(tdb, tr, vr, rh, met, t_running_mean) {
+  tdb = Array.isArray(tdb) ? tdb : [tdb];
+  tr = Array.isArray(tr) ? tr : [tr];
+  vr = Array.isArray(vr) ? vr : [vr];
+  met = Array.isArray(met) ? met : [met];
+  rh = Array.isArray(rh) ? rh : [rh];
+  t_running_mean = Array.isArray(t_running_mean)
+    ? t_running_mean
+    : [t_running_mean];
+
+  //assume all parameters have same length, and it should have same length when use this function
+  const met_adapted = met.map(
+    (metValue, index) => metValue - (0.234 * t_running_mean[index]) / 58.2,
+  );
+
+  const clo_adapted = met_adapted.map((metValue, index) =>
+    Math.pow(
+      10,
+      -0.17168 -
+        0.000485 * t_running_mean[index] +
+        0.08176 * metValue -
+        0.00527 * t_running_mean[index] * metValue,
+    ),
+  );
+
+  const pmv_res = tdb.map((tdbValue, i) =>
+    pmv_ppd_optimized(
+      tdbValue,
+      tr[i],
+      vr[i],
+      rh[i],
+      met_adapted[i],
+      clo_adapted[i],
+      0,
+    ),
+  );
+
+  const ts = met_adapted.map(
+    (metValue) => 0.303 * Math.exp(-0.036 * metValue * 58.15) + 0.028,
+  );
+
+  const l_adapted = pmv_res.map((pmv_res_Value, i) => pmv_res_Value / ts[i]);
+
+  return l_adapted.map((lValue, index) =>
+    parseFloat(
+      (
+        1.484 +
+        0.0276 * lValue -
+        0.9602 * met_adapted[index] -
+        0.0342 * t_running_mean[index] +
+        0.0002264 * lValue * t_running_mean[index] +
+        0.018696 * met_adapted[index] * t_running_mean[index] -
+        0.0002909 * lValue * met_adapted[index] * t_running_mean[index]
+      ).toFixed(3),
+    ),
+  );
+}
+
+function pmv_ppd_optimized(tdb, tr, vr, rh, met, clo, wme) {
```

Comment:
why is this function here? (should not include double implementations in multiple PR)
also why is it called optimized?
what is optimized about it? (aka this does not make sense in the javascript version) (the python version was using a vectorized technique, also using JIT and caching) (here we are not doing any of that so no point in calling it optimized)

Reply:
so here, should I use the function pmv instead?